pytorch_lightning/plugins/training_type/tpu_spawn.py

```diff
@@ -0,0 +1,184 @@
+import io
+import os
+from pytorch_lightning.core.lightning import LightningModule
+import torch
+from pytorch_lightning.utilities.distributed import rank_zero_only
+from pytorch_lightning.utilities.seed import seed_everything
+from typing import Any, Dict, Optional
+from pytorch_lightning.plugins.training_type.ddp_spawn import DDPSpawnPlugin
+from pytorch_lightning.utilities import _TPU_AVAILABLE, rank_zero_warn,
+
+from pytorch_lightning.utilities.apply_func import move_data_to_device
+
+if _TPU_AVAILABLE:
+    import torch_xla
+    import torch_xla.core.xla_model as xm
+    import torch_xla.distributed.parallel_loader as xla_pl
+    import torch_xla.distributed.xla_multiprocessing as xmp
+
+class TPUSpawnPlugin(DDPSpawnPlugin):
+    def __init__(self, parallel_devices, num_nodes=1, **kwargs: Dict[str, Any]):
+
+        parallel_devices = [xm.xla_device(device) if isinstance(device, int) else device for device in parallel_devices]
+        super().__init__(parallel_devices, num_nodes=num_nodes, cluster_environment=None, sync_batchnorm=False, **kwargs)
+        self.tpu_local_core_rank = 0
+        self.start_method = None
+
+    @property
+    def distributed_sampler_kwargs(self):
+        return dict(num_replicas=xm.xrt_world_size(), rank=xm.get_ordinal())
+
+    def process_dataloader(self, dataloader):
+        device = xm.xla_device(self.trainer.tpu_id)
```

Comment:
dont require passing in `self.trainer.tpu_id` here.

Reply:
But does it hurt?

Reply:
`self.trainer.tpu_id` will always have to be `None` here. Else assigning a device manually within spawn will result in an error.

Reply:
Got it!